crates/network/src/ethereum/mod.rs

```diff
@@ -35,6 +35,21 @@ impl ReceiptResponse for alloy_rpc_types_eth::TransactionReceipt {
     fn contract_address(&self) -> Option<alloy_primitives::Address> {
         self.contract_address
     }
+
+    /// ## Note
+    ///
+    /// Caution must be taken when using this method for deep-historical
+    /// receipts, as it may not accurately reflect the status of the
+    /// transaction. The transaction status is not knowable from the receipt
+    /// for transactions before [EIP-658].
+    ///
+    /// This can be handled using [`TxReceipt::status_or_post_state`].
+    ///
+    /// [EIP-658]: https://eips.ethereum.org/EIPS/eip-658
+    /// [`TxReceipt::status_or_post_state`]: alloy_consensus::TxReceipt::status_or_post_state
```

Comment:
duplicating the docs is not necessary imo

Reply:
i'd like the docs to be on every place the user may encounter this (potentially confusing) behavior

Reply:
trait impls never need docstrings tho :)

Reply:
I got it, removed the docs from the trait impls: https://github.com/alloy-rs/alloy/pull/846/commits/b42dce55848adf6e5e5342d7e50083087729c5da